packages/blocks/src/file-input/index.js

```diff
@@ -60,7 +59,8 @@ const FileInput = ( { attributes, className } ) => {
 			</FormInputWrapper.Label>
 			<FormFileInput
 				attributes={ attributes }
-				inputProps={ inputProps }
+				files={ fieldValue }
+				onChange={ ( event ) => onUpdate( event.target.files ) }
```

Comment:
The `event -> value` mapping should be done in our input components. So here it's just `onChange={ onUpdate }` or `onChange={ setFieldValue }`.